kubectl-cloudflow/deploy/deploy_test.go

```diff
@@ -1,103 +1,273 @@
+			console-egress = outp/carly.output.AggregateRecordEgress
+			error-egress = outp/carly.output.InvalidRecordEgress
+		}
+		connections {
+			cdr-generator1.out = [merge.in-0]
+			cdr-generator2.out = [merge.in-1]
+			cdr-ingress.out = [merge.in-2]
+			merge.valid = [cdr-aggregator.in]
+			merge.invalid = [error-egress.in]
+			cdr-aggregator.out = [console-egress.in]
+		}
+	}`
 }
 
-func commandLineForConfiguration() []string {
-	return []string{
-		`valid-logger.log-level=warning`,
-		`valid-logger.msg-prefix=test`,
-		`valid-logger.msg-prefix-2="test"`,
-		`valid-logger.msg-prefix-3=20`,
-		`valid-logger.msg-prefix-4=3.15`,
-		`valid-logger.msg-prefix-5=true`,
-		`valid-logger.msg-prefix-6="20 minutes"`,
-		`valid-logger.msg-prefix-7="20M"`}
+func createImageDigests() []string {
+	digests := make([]string, 3)
+
+	digests[0] = "sha256:98bded501c9e7e9101cbc0971cc529e9d2240459924b30b44b2ea1976c328a46"
+	digests[1] = "sha256:b5eb3e5d5bb245ef322ac9ac1698c7ee7543707c20184a3e94824afc429567e9"
+	digests[2] = "sha256:567e1fb63145a5f267894d70ca49d38573655c6b6d97ca64fb41402d2aede208"
+	return digests
+}
+
+func getBlueprintConfig(contents string) (*configuration.Config, error) {
+	config := configuration.ParseString(contents)
+	return config, nil
 }
 
-func Test_SplitConfigurationParameters(t *testing.T) {
-	result := SplitConfigurationParameters([]string{
-		`a="b"`,
-		"key=ddd",
-		`some.key="some, value"`,
-		`some.key2=c29tZSBzdHJpbmc9PQ==`,
-		`some.key3="some text passed to a streamlet"`,
-		`some.key4=post Cobra processed, unquoted string`})
-
-	assert.NotEmpty(t, result)
-	assert.Equal(t, result["a"], "b")
-	assert.Equal(t, result["key"], "ddd")
-	assert.Equal(t, result["some.key"], "some, value")
-	assert.Equal(t, result["some.key2"], "c29tZSBzdHJpbmc9PQ==")
-	assert.Equal(t, result["some.key3"], "some text passed to a streamlet")
-	assert.Equal(t, result["some.key4"], "post Cobra processed, unquoted string")
-
-	empty := SplitConfigurationParameters([]string{})
-	assert.Empty(t, empty)
+func filter(ss []streamletRef, val string, test func(streamletRef, string) bool) (ret []streamletRef) {
+	for _, s := range ss {
+		if test(s, val) {
+			ret = append(ret, s)
+		}
+	}
+	return
 }
 
-func Test_ValidateConfigurationAgainstDescriptor(t *testing.T) {
+func unique(stringSlice []string) []string {
+	keys := make(map[string]bool)
+	list := []string{}
+	for _, entry := range stringSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
 
-	applicationConfiguration := domain.TestApplicationDescriptor()
+var _ = Describe("Deploy", func() {
+	var (
+		bp             blueprint
+		streamletIDs   []string
+		streamletIDMap map[string]bool
+	)
 
-	var spec domain.CloudflowApplicationSpec
-	json.Unmarshal([]byte(applicationConfiguration), &spec)
+	BeforeEach(func() {
+		config, err := getBlueprintConfig(createBlueprint())
+		Ω(err).ShouldNot(HaveOccurred())
```

Comment:
what is the `Ω` for?

Reply:
oh that must be gomega syntax? would not have expected that in golang...

Reply:
yeah .. replaced with sanity .. (my Haskell instincts took over momentarily - sorry)

Reply:
lol